src/test/clojure/clara/test_durability.clj

```diff
@@ -1,163 +1,252 @@
-
-    (is (has-fact? @rule-output (->Temperature 10 "MCI")))))
-
-
-(deftest test-restore-fired-activation
-  (let [rule-output (atom nil)
-        cold-rule (dsl/parse-rule [[Temperature (< temperature 20)]]
-                                  (reset! rule-output ?__token__))
-
-        session-state (-> (mk-session [cold-rule])
-                          (insert (->Temperature 10 "MCI"))
-                          (fire-rules)
-                          (d/session-state))]
-
-    ;; Rule fired previously, so the output should be in place.
-    (is (has-fact? @rule-output (->Temperature 10 "MCI")))
-
-    ;; Clear the output and restore the session. It should already be activated,
-    ;; so the output is not updated again.
-    (reset! rule-output nil)
-
-    (-> (mk-session [cold-rule])
-        (restore-session session-state)
-        (fire-rules))
-
-    (is (nil? @rule-output))))
-
-(deftest test-restore-accum-result
-  (let [lowest-temp (acc/min :temperature :returns-fact true)
-        coldest-query (dsl/parse-query [] [[?t <- lowest-temp from [Temperature]]])
-
-        session (-> (mk-session [coldest-query])
-                    (insert (->Temperature 15 "MCI")
-                            (->Temperature 10 "MCI")
-                            (->Temperature 80 "MCI"))
-                    (fire-rules))
-
-        session-state (d/session-state session)
-
-        restored-session (-> (mk-session [coldest-query])
-                             (restore-session session-state))]
-
-    ;; Accumulator returns the lowest value.
-    (is (= #{{:?t (->Temperature 10 "MCI")}}
-           (set (query session coldest-query))
-           (set (query restored-session coldest-query))))))
-
-(deftest test-restore-truth-maintenance
-  (let [cold-rule (dsl/parse-rule [[Temperature (< temperature 20) (= ?temp temperature)]]
-                                  (insert! (->Cold ?temp)))
-
-        cold-query (dsl/parse-query [] [[?cold <- Cold]])
-
-        empty-session (mk-session [cold-rule cold-query])
-
-        persist-and-restore (fn [session]
-                              (restore-session empty-session
-                                               (d/session-state session)))
-
-        cold-result {:?cold (->Cold 10)}]
-
-    (let [session (-> empty-session
-                      (insert (->Temperature 10 "MCI"))
-                      (fire-rules))
-
-          session-state (d/session-state session)
+            [schema.test :as st]
+            [clojure.java.io :as jio]
+            [clojure.test :refer :all])
+  (:import [clara.rules.testfacts
+            Temperature]))
+
+(use-fixtures :once st/validate-schemas)
+
+(defrecord LocalMemorySerializer [holder]
+  d/IWorkingMemorySerializer
+  (serialize-facts [_ fact-seq]
+    (reset! holder fact-seq))
+  (deserialize-facts [_]
+    @holder))
+
+(def rulebase-data (java.io.File/createTempFile "rulebase" ".dat"))
+(def session-data (java.io.File/createTempFile "session" ".dat"))
```

Comment:
Good point.  I can completely remove these and just have local bindings on ByteArrayInputStream and ByteArrayOutputStream's actually.  I'll update this PR with those changes shortly.